packages/analytics-js-integrations/src/integrations/Mixpanel/util.js

```diff
@@ -239,6 +239,39 @@ const getConsolidatedPageCalls = config =>
     ? config.consolidatedPageCalls
     : true;
 
+/**
+ * Generates a custom event name for a page or screen.
+ *
+ * @param {Object} message - The message object
+ * @param {string} userDefinedEventTemplate - The user-defined event template to be used for generating the event name.
+ * @throws {ConfigurationError} If the event template is missing.
+ * @returns {string} The generated custom event name.
+ * @example
+ * const userDefinedEventTemplate = "Viewed {{ category }} {{ name }} Page";
+ * const message = {name: 'Home', properties: {category: 'Index'}};
+ * output: "Viewed Index Home Page"
+ */
+const generatePageCustomEventName = (message, userDefinedEventTemplate) => {
+  let eventName = userDefinedEventTemplate;
+
+  // find {{ category }} surrounded by whitespace characters and replace it with a single whitespace character
+  if (!isDefinedAndNotNull(message.properties?.category)) {
+    eventName = eventName.replace(/\s{{\s*category\s*}}\s/g, ' ');
+  }
+
+  // find {{ name }} surrounded by whitespace characters and replace it with a single whitespace character
+  if (!isDefinedAndNotNull(message.name)) {
+    eventName = eventName.replace(/\s{{\s*name\s*}}\s/g, ' ');
+  }
+
+  // Replace placeholders with actual values
+  eventName = eventName
+    .replace('{{ category }}', message.properties?.category)
+    .replace('{{ name }}', message.name);
```

Comment:
Should also use the same regex format here?
`{{\s*category\s*}}` instead of `{{ category }}`

Reply:
Also, you can replace with actual values in the else blocks of the above if blocks.

if (property is defined) {
 replace the values
} else {
 check for surrounding whitespace and fuse them
}